src/test/java/com/github/quadflask/smartcrop/HaarCascadeFaceDetector.java

```diff
@@ -0,0 +1,51 @@
+package com.github.quadflask.smartcrop;
+
+import org.bytedeco.javacpp.Loader;
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Rect;
+import org.bytedeco.javacpp.opencv_core.RectVector;
+
+import org.bytedeco.javacpp.opencv_java;
+import org.bytedeco.javacpp.opencv_objdetect.CascadeClassifier;
+import org.bytedeco.javacv.Java2DFrameUtils;
+
+import java.awt.image.BufferedImage;
+
+import static org.bytedeco.javacpp.opencv_imgproc.COLOR_BGR2GRAY;
+import static org.bytedeco.javacpp.opencv_imgproc.cvtColor;
+import static org.bytedeco.javacpp.opencv_imgproc.equalizeHist;
+
+public class HaarCascadeFaceDetector implements FaceDetector {
+
+    static {
+        Loader.load(opencv_java.class);
+
+        CascadeClassifier classifier = new CascadeClassifier();
+        classifier.load("src/test/resources/haarcascade_frontalface_default.xml");
+        faceCascade = classifier;
+    }
+
+    private static CascadeClassifier faceCascade;
+
+    @Override
+    public void detect(BufferedImage input, FaceConsumer consumer) {
+        Mat frame = Java2DFrameUtils.toMat(input);
+
+        Mat grayFrame = new Mat();
+        cvtColor(frame, grayFrame, COLOR_BGR2GRAY);
```

Comment:
Not sure if conversion to gray is required, but the examples I've seen using this detector have converted the source image to gray. I can't see much difference in the detection performance either way.